### DISCUSSION TITLE
Customisable column ordering, visibility and pinning modal

### DISCUSSION THREAD

**Discussion by kaushalendra-pandey** [KevinVandy/material-react-table#940] [General]
The modal that comes by default for setting column properties suffice the basic use-cases but lacks some configuration like options to select if we want to show the pinning icons or if we want to show visibility checkboxes etc.
Also the modification is also not possible. For example, i wanted to add a search bar in the modal to let them search the column as the number of column can be more.
Can we make this modal more customisable and expose API to control the CSS of this modal?

<img width="343" alt="image" src="https://github.com/KevinVandy/material-react-table/assets/67643722/6daed5b7-4bd6-4e5f-b88f-6bdd173ee25d">

**Comment by KevinVandy** [comment]
As I said in the column hiding docs, this is one of the few opinionated parts of MRT. Instead of making a dozen more props to allow customization of this, I made easier for you to just replace it with your own button and menu. And table and column instance APIs are available for you to make whatever you need. You can even import some of the smaller MRT subcomponents that make up the show/hide menu in your own implementation.
